src/qutip_qip/decompose/decompose_general_qubit_gate.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+import cmath
+
+from qutip import Qobj
+from qutip_qip.decompose._utility import (
+    check_gate,
+    MethodError,
+)
+
+import warnings
+from qutip_qip.circuit import Gate
+from qutip_qip.operations import controlled_gate
+
+from .decompose_single_qubit_gate import decompose_one_qubit_gate
+
+# for unknown labels in two level gates
+warnings.filterwarnings("ignore", category=UserWarning)
+
+
+def _decompose_to_two_level_arrays(input_gate, num_qubits):
```

Comment:
The output of this has been checked for two qubits in a previous PR. The change here has been to find the `index_list` first. 

The output has not been fully checked. The length of the output was checked and not their product. 

The output will be checked once an easier way to define gate objects for these arrays has been added. 

- 2 qubits - array length is 6
- 3 qubits - array length is 28
- 4 qubits  - array length is 120